### DISCUSSION TITLE
Very easy stupid question about linking a domain name

### DISCUSSION THREAD

**Discussion by crowscrows** [rr-/szurubooru#579] [Q&A] [answered]
I followed every instructions, I opened up a port both on my machine and on my router (which is 8080) and I managed to run the server on my machine with an address that looks like this 12.34.567.890:8080
It works well, I can send the address to my friends and everything works as intended. 

But how do I go about linking it to a domain name? I bought a domain from google domains and I linked it with my IP (without the 8080 port) then I changed the url on the .env file for the url of the domain I bought, but it still doesn't link correctly.

What am I missing?

**Comment by TiredSysOp** [comment]
Post config and DNS entries
Check you can access Szurubooru externally on port 80
Check DNS records

**Reply by TiredSysOp** [reply]
Create an A record for the root of your domain pointing to your IP. If you can access it externally with your IP then it's a DNS issue.

**Reply by crowscrows** [reply]
I can access it with my IP and port, but not my ip only (without port)

My DNS entries are very simple, It's an A record with my IP address that's it.

what config files would you like to see?

**Comment by noirscape** [comment, marked as answer]
You need to open port 80 and have szurubooru (or a different webserver that reverse proxies to szurubooru) listen on it. Check "preparing for production" in install.md if you are using nginx.

After that, make sure you have the `A`/`AAAA` record (`AAAA` only if you have ipv6) set up properly. From your description, right now you should be able to access your booru on `domain:8080` (so just do `:8080` at the end of the domain name). It might take a day or two for your DNS changes to propagate, so be aware of that.

**Reply by crowscrows** [reply]
adding the port number at the end on the domain name works, I didn't know that, thanks! 

Is there any downside to not doing any reverse proxying? I am currently not using nginx or any other websoftware aside from the ones found on the installation instructions

**Reply by noirscape** [reply]
Honestly the main problem is that you're letting the files get served by waitress instead of nginx. It usually doesn't matter a *whole* lot, but if you ever get serious traffic, it could bring down the server since waitress isn't really designed for that sort of thing.

The other reason to do reverse proxying is https, which is again, not required... unless you want to make use of a bunch of modern web technologies (not inherently needed for szuru which doesn't make use of them), but your browser is going to say that your site is insecure and might complain about you entering your password (since anyone who connects to your site can be spied on - that's what HTTPS is meant to protect).

YMMV on how dangerous that is.

**Comment by crowscrows** [comment]
I just realised what I was trying to do is hide the port number, and the solution to this is just setting my port to 80 instead of 8080! Works like I want to! Thanks for the help!
